nixos/modules/config/pulseaudio.nix

```diff
@@ -135,6 +177,7 @@ in {
 
     (mkIf cfg.enable {
       environment.systemPackages = [ cfg.package ];
+      environment.sessionVariables = moduleEnvVars;
```

Comment:
I mean wrapping pulseaudio, but not itself -- rather as a part of a wrapper (see deadbeef, pidgin, dwarf-therapist etc.) The main idea is:

```
symlinkJoin {
  name = "pulseaudio-with-packages";
  paths = [ pulseaudio ] ++ cfg.packages;
  buildInputs = [ makeWrapper ];
  postBuild = ''
    wrapProgram $out/bin/pulseaudio \
      --set PA_ALSA_PATHS_DIR ... \
      --set PA_ALSA_PROFILE_SETS_DIR ....
  '';
}
, and this goes directly to `systemPackages`. This drops need for system-wide environment variables and may make the whole thing simpler (because most of `buildCommand` code above should not be needed this way -- if I'm not mistaking anything!).